### PR TITLE
Develop js_herbivores_and_carnivores

### PR DESCRIPTION
**Pull request by sergey-lipikhin**

### CODE REVIEW

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -1,15 +1,52 @@
...
+    if (value >= 0) {
+      this._health = value;
+    } else {
+      this._health = 0;
+    }
+  }
+
+  get health() {
+    return this._health;
+  }
```
Comment (mstrilec) [inline src/herbivoresAndCarnivores.js:21]: you don't need it at all.

**Review by mstrilec — CHANGES_REQUESTED** [review]

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -1,15 +1,40 @@
...
+  bite(prey) {
+    if (
+      prey instanceof Herbivore
+      && !prey.hidden
+    ) {
+      prey.health -= 50;
+
+      if (prey.health === 0) {
+        Animal.alive = Animal.alive.filter((animal) => animal !== prey);
+      }
```
Comment (OleksiiBlezniuk) [inline src/herbivoresAndCarnivores.js:35]: Let's avoid nested if's

Comment (OleksiiBlezniuk) [reply src/herbivoresAndCarnivores.js:35]: And your solution will be more scalable if you'll check if health is greater or equal than 0
What if we'll change damage or health?

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

File: src/herbivoresAndCarnivores.js [hunk 3]
```diff
@@ -1,15 +1,56 @@
...
+
+    Animal.alive.push(this);
+  }
+
+  set health(value) {
+    if (value >= 0) {
+      this._health = value;
+    } else {
+      this._health = 0;
+    }
```
Comment (sergii-nosachenko) [inline src/herbivoresAndCarnivores.js:16]: Good place for ternary operator.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

**Review by sergii-nosachenko — APPROVED** [review]
Good work 👍

**Pull request closed without merging**